monkey/monkey_island/cc/services/utils/config_encryption.py

```diff
@@ -0,0 +1,56 @@
+import base64
+import io
+import json
+import logging
+from typing import Dict, Union
+
+import pyAesCrypt
+
+from common.utils.exceptions import InvalidCredentialsError, NoCredentialsError
+
+BUFFER_SIZE = pyAesCrypt.crypto.bufferSizeDef
+
+logger = logging.getLogger(__name__)
+
+
+def encrypt_config(config: Dict, password: str) -> str:
+    plaintext_config_stream = io.BytesIO(json.dumps(config).encode())
+    ciphertext_config_stream = io.BytesIO()
+
+    pyAesCrypt.encryptStream(
+        plaintext_config_stream, ciphertext_config_stream, password, BUFFER_SIZE
+    )
+
+    ciphertext_b64 = base64.b64encode(ciphertext_config_stream.getvalue())
+    logger.info("Configuration encrypted.")
+
+    return ciphertext_b64.decode()
+
+
+def decrypt_config(cyphertext: Union[str, dict], password: str) -> Dict:
+    if not password:
+        raise NoCredentialsError
+
+    cyphertext = base64.b64decode(cyphertext)
```

Comment:
Does this barf if we give it a `dict`? See above comment on the `Union` typehint for context.